Refactore logic

Qmemorais opened the pull request.

Qmemorais committed:
Create GetRequestFromURL.cs

Create new class to get response from URL and some values from response

Qmemorais committed:
Delete GetResponseFromURL.cs

Qmemorais committed:
Update GetSettingFromURL.cs

remove method to get ContentType to class which work with response

Qmemorais committed:
Update LogicScanByHTML.cs

add method if url contains special symbols. To remove  part of url which are don't work

Qmemorais committed:
Update LogicScanBySitemap.cs

remove try/catch. remove response method to Request class

Qmemorais committed:
Update MainLogic.cs

remove try/catch

Qmemorais committed:
Update OutputList.cs

create new method to get exist lista nd then write it to field

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+            if (GetStatusCode(url) == HttpStatusCode.OK)
+            {
+                var response = GetResponse(url);
+                var read = new StreamReader(response.GetResponseStream(), Encoding.Default, true, 8192);
```
Rename variable `read` using noun instead of verb

Heorhii-Svitenko-Ukad replied:
Better to get encoding from response object, and use another (simplier) StreamReader constructor override

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+        {
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+            return (HttpWebResponse)request.GetResponse();
+        }
```
It makes no sense to extract the creation of a web request and its invocation

Heorhii-Svitenko-Ukad replied:
It seems that there also should be a `try/catch` block

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+
+namespace TestURLS.UrlLogic
+{
+    public class GetRequestFromURL
```
`GetRequestFromURL` can be a good name for a method, not a class

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+            catch(WebException e)
+            {
+                HttpWebResponse httpResponse = (HttpWebResponse)e.Response;
+                return httpResponse.StatusCode;
```
It can be a good idea to place a logging there (but not required)

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+            if (GetStatusCode(url) == HttpStatusCode.OK)
+            {
+                var response = GetResponse(url).ContentType;
+                contentType = response.IndexOf("text/html") != -1;
```
Better to use `.Contains()` instead of `.IndexOf()`

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+
+        public virtual string GetBodyFromURL(string url)
+        {
+            var HTMLtxt = "";
```
Better to use `string.Empty` construction in place of `""`

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+        public virtual string GetSitemapFromURL(string url)
+        {
+            string line;
+            string sitemap = "";
```
Better to use `string.Empty` construction in place of `""`

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+            if (GetStatusCode(url) == HttpStatusCode.OK)
+            {
+                var response = GetResponse(url);
+                var reader = new StreamReader(response.GetResponseStream(), Encoding.Default, true, 8192);
```
Better to get encoding from response object, and use another (simplier) StreamReader constructor override

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+                var reader = new StreamReader(response.GetResponseStream(), Encoding.Default, true, 8192);
+                while ((line = reader.ReadLine()) != null)
+                {
+                    if (line.IndexOf("Sitemap: ") != -1)
```
Better to use `.Contains()` instead of `.IndexOf()`

Heorhii-Svitenko-Ukad commented on TestURLS.UrlLogic/GetRequestFromURL.cs:
```diff
@@ -0,0 +1,75 @@
+            return HTMLtxt;
+        }
+
+        public virtual string GetSitemapFromURL(string url)
+        {
+            string line;
+            string sitemap = "";
+            if (GetStatusCode(url) == HttpStatusCode.OK)
+            {
+                var response = GetResponse(url);
+                var reader = new StreamReader(response.GetResponseStream(), Encoding.Default, true, 8192);
+                while ((line = reader.ReadLine()) != null)
+                {
+                    if (line.IndexOf("Sitemap: ") != -1)
+                    {
+                        sitemap = line[9..];
+                    }
+                }
+                response.Close();
+            }
+            return sitemap;
+        }
```
I guess, we can separate getting the content of the response and extracting the sitemap url

Qmemorais committed:
Rename GetSettingFromURL

rename class/ add method to understand if url is valid and usable

Qmemorais committed:
Relocate Output class

remove class to Console project because this is output strings

Qmemorais committed:
Rename class with Response methods

Qmemorais committed:
Update TestURLS.urlLogic.csproj

Add new library to Parse html page like XML Doc

Qmemorais committed:
Update LogicScanByHTML.cs

Remove Regex match to find href="". Use XML Doc instead of Regex.Match. Make new method to get Valid URL in SettingURL class

Qmemorais committed:
Update LogicScanBySitemap.cs

update call method to get response. get status. if it`s ok then get values from Sitemap by using XML Documents

Qmemorais committed:
Update MainLogic.cs

Make method to get List with existing URLs in ONE scan. make fields like properties to get it from ConsoleApp

Qmemorais committed:
Update URLSettings.cs

make new method to avoid If in If

Qmemorais committed:
Update MainLogic.cs

create method to get time for response from url and return to ConsoleApp

Qmemorais committed:
Update LogicScanBySitemap.cs

make spaces between lines to make more readable

Qmemorais committed:
Update LogicScanByHTML.cs

make new method which return bool type if url have valid view

Qmemorais committed:
Update HttpLogic.cs

remove method which already exist in class

Qmemorais committed:
Update ConsoleInterface.cs

remove try/catch block

Qmemorais committed:
Update OutputToConsole.cs

make code more readable. remove method to MainLogic which return Dictionary with urls and time of response

Qmemorais committed:
Create UrlTimeModel.cs

create class-model to save time from response and link

Qmemorais committed:
Update URLSettings.cs

refactore code to more readable

Qmemorais committed:
Create Time.cs

create class and method to get time from response and save it

Qmemorais committed:
Update MainLogic.cs

update return type of GetResults method

Qmemorais committed:
Update LogicScanBySitemap.cs

refactore code to more readable

Qmemorais committed:
Update LogicScanByHTML.cs

refactore code to more readable

Qmemorais committed:
Update HttpLogic.cs

remove method to get content type. add more "if" o understand is link is good

Qmemorais committed:
Update OutputToConsole.cs

add interpolation

Qmemorais committed:
Update ConsoleInterface.cs

Qmemorais committed:
Update .gitignore

add .gitignore new project with models

Qmemorais committed:
Create new project Models

create new project for models:
url+where if found
url+time of response

Qmemorais committed:
Update ConsoleInterface.cs

Qmemorais committed:
Update OutputToConsole.cs

because of create new model class- update return type and method to output in console

Qmemorais committed:
Add references in projects

add references for new project of Models

Qmemorais committed:
Update HttpLogic.cs

add type of exception

Qmemorais committed:
Update LogicScanByHTML.cs

update return type of value. add new method to remove part with "#" if link doesn`t exist

Qmemorais committed:
Update LogicScanBySitemap.cs

remove part of scan robot.txt file.
add new method to create new class from model UrlWithWhereScan

Qmemorais committed:
Rename Time.cs

Qmemorais committed:
Delete UrlTimeModel.cs

remove this model class to new project with models

Qmemorais committed:
Update TestURLS.urlLogic.csproj

add reference for new project with models

Qmemorais committed:
Update MainLogic.cs

update method which find links from one bot two

Qmemorais committed:
Make new type of fields

make fileds type var

Qmemorais committed:
update method to get distinctList

Qmemorais committed:
Update LogicScanByHTML.cs

create new method to get matches and add it to generic list

Qmemorais committed:
Update ConsoleInterface.cs

Qmemorais committed:
Update OutputToConsole.cs

Qmemorais committed:
Update URLSettings.cs

Update some names of method. create new method if links from sitemap is not exactly that from scanWeb

Qmemorais committed:
Update TimeTracker.cs

Add fields which helps to get links which exists in sitemap and web and only in sitemap

Qmemorais committed:
Remove project Model. Create folder

Qmemorais committed:
Update MainLogic.cs

add method to get ResponseTime with link. Update method to get different links

Qmemorais committed:
Rename fields in Sitemap/Html

Qmemorais committed:
Rename fields in Http/IConsole

Qmemorais committed:
Update OutputToConsole.cs

add new method to write only links with time. add new method to write only coun of links
Calles from ConsoleInterface

Qmemorais committed:
Rename ConsoleProject

rename project. add condition if links from sitemap don`t exist

Qmemorais committed:
Update OutputToConsole.cs

recall method with only two upper symbols

Qmemorais committed:
Update HttpLogic.cs

make code more readable. add some spaces between blocks

Qmemorais committed:
Rename DomenName to domainName

Qmemorais committed:
Update TimeTracker.cs

remove unnecesary fields

Qmemorais committed:
Update MainLogic.cs

make condition instead of Object.Equal make string.Equal

Qmemorais committed:
Create dependency on interfaces

make all classes from logic and console dependency on interfaces to avoid fat coupling

Qmemorais committed:
Update locate of interface IConsoleInOut

Qmemorais committed:
Create ILogicToConsole.cs

Qmemorais committed:
Create IOutputToConsole.cs

create interface for OutputToConsole public methods

Qmemorais committed:
Create IHttpLogic.cs

create interface for HttpLogic public methods

Qmemorais committed:
Create ILogicScanByHtml.cs

create interface for LogicScanBuHtml public methods

Qmemorais committed:
Create ILogicScanBySitemap.cs

create interface for LogicScanBySitemap public methods

Qmemorais committed:
Create IMainLogic.cs

create interface for MainLogic public methods

Qmemorais committed:
Create ITimeTracker.cs

create interface for TimeTracker public methods

Qmemorais committed:
Create IUrlSettings.cs

create interface for UrlSettings public methods

Qmemorais committed:
Create method to add DI

add reference on library to use DI
create method to add coupling between interfaces and classes

Qmemorais committed:
Split ConfigureServices to projects

add configurationServices to each project with his classes and interfaces

Qmemorais committed:
remove interfaces from some classes

Qmemorais committed:
make services ICollection

Qmemorais committed:
Update tets for Console Logic.cs

Qmemorais committed:
Update level of security

Qmemorais committed:
Update MainLogicTests.cs

Update tests to getResults With exist Sitemap and NoExist Sitemap
